### PR TITLE
binding: support parse URL parameters into struct fields

### PR DESCRIPTION
**Pull request by xmh19936688**
support args in url like '/path/:ID ' can be bind into struct

### CODE REVIEW

**Comment by unknwon** [comment]
Thanks for the PR!

Unfortunately, I do not think it fits the use cases of this package. It is used for form/content binding.

**Comment by xmh19936688** [comment]
How about make the funcs variable? Then developers can customize them as macaron.ReturnHandler.
Like this:
```diff
+var (
+	FormBindHandler          BindHandler = Form
+	MultipartFormBindHandler BindHandler = MultipartForm
+	JsonBindHandler          BindHandler = Json
+)

func bind(ctx *macaron.Context, obj interface{}, ifacePtr ...interface{}) {
	contentType := ctx.Req.Header.Get("Content-Type")
	if ctx.Req.Method == "POST" || ctx.Req.Method == "PUT" || ctx.Req.Method == "PATCH" || ctx.Req.Method == "DELETE" {
		switch {
		case strings.Contains(contentType, "form-urlencoded"):
-			ctx.Invoke(Form(obj, ifacePtr...))
+			ctx.Invoke(FormBindHandler(obj, ifacePtr...))
		case strings.Contains(contentType, "multipart/form-data"):
-			ctx.Invoke(MultipartForm(obj, ifacePtr...))
+			ctx.Invoke(MultipartFormBindHandler(obj, ifacePtr...))
		case strings.Contains(contentType, "json"):
-			ctx.Invoke(Json(obj, ifacePtr...))
+			ctx.Invoke(JsonBindHandler(obj, ifacePtr...))
		default:
			var errors Errors
			if contentType == "" {
				errors.Add([]string{}, ERR_CONTENT_TYPE, "Empty Content-Type")
			} else {
				errors.Add([]string{}, ERR_CONTENT_TYPE, "Unsupported Content-Type")
			}
			ctx.Map(errors)
			ctx.Map(obj) // Map a fake struct so handler won't panic.
		}
	} else {
+		ctx.Invoke(Form(obj, ifacePtr...))
+		ctx.Invoke(FormBindHandler(obj, ifacePtr...))
	}
}
```

**Comment by unknwon** [comment]
Hmm, thanks for the brainstorming! I'm very cautious to add global variables recently. 

What about adding another top level function `binding.URL(...)`, which specifically binds URL parameters to a struct?

File: binding.go [hunk 1]
```diff
@@ -112,6 +112,25 @@ func Bind(obj interface{}, ifacePtr ...interface{}) macaron.Handler {
...
+		var errors Errors
+
+		ensureNotPointer(obj)
+		obj := reflect.New(reflect.TypeOf(obj))
+
+		val := obj.Elem()
+		for k, v := range ctx.AllParams() {
+			field := val.FieldByName(k[1:])
+			if field.IsValid() {
+				field.SetString(v)
```
Comment (unknwon) [inline binding.go]: I think assuming the struct field type is always `string` is dangerous (i.e. could lead to panic?) Instead, I believe we should call `setWithProperType` for each URL parameter we found, see examples: https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/go-macaron/binding%24%40master++%3D+setWithProperType+-file:_test%5C.go%24+&patternType=literal

**Comment by unknwon** [comment]
Thanks for the update, one last issue commented :)

**Pull request merged**